jwst/lib/basic_utils.py

```diff
@@ -1,9 +1,32 @@
 """General utility objects"""
 
+from stdatamodels.jwst import datamodels
 from stdatamodels.jwst.datamodels import dqflags
 import numpy as np
 
 
+def use_datamodel(input, ramp_model=False):
+    """Determine if input is a datamodel, if so return it, else open it
+
+    Parameters
+    ----------
+    input : string or datamodel
+        Either the name of the file to open or a datamodel
+
+    Returns
+    -------
+    model : datamodel
+        The datamodel object
+    """
+    if isinstance(input, datamodels.JwstDataModel):
+        model = input
+        if ramp_model:
+            model = datamodels.RampModel(model)
+    else:
+        model = datamodels.open(input)
```

Comment:
What if someone gives a file name string as input AND wants to force it to be returned as a `RampModel`, i.e. `ramp_model=True`? I think this `else` branch should allow for that case.

Reply:
Agreed.

And also, it might be more generally useful to allow this function to take a datamodel class as an optional argument, instead of just allowing RampModel specifically, via the boolean.

Reply:
The datamodels accept a filename or a model as an "init" option . For example both `RampModel("some_filename.fits")` and `RampModel(model)` return a `RampModel`. This is equivalent to `use_datamodel(model, True)` or `use_datamodel("some_filename.fits", True)` for this utility function so I don't think we need to support that feature and can instead use `RampModel(input)`.

Reply:
good idea, I added an option to provide whatever model needed

Reply:
This option effectively undid some of your other changes (and added extra clones in some situations). Take [lastframe_step](https://github.com/spacetelescope/jwst/pull/8588/files#diff-33c77eb6d332c3ca68e92903ac6d92bfe932ee09fb133e25442aa91da822a866R21) as an example. On main the code is:
```
with datamodels.RampModel(input) as input_model:
```
and with this PR the code is:
```
with use_datamodel(input, model_class=datamodels.RampModel) as input_model:
```
When `input` is already a `RampModel` (as is the case for a run of `detector1`). The old code would create a new "cloned" `RampModel`. The new code now also makes a clone (when `model_class(model)` is called). In both cases an extra clone is made and the input is not modified (the clone is).
When `input` is some other non-`RampModel` both the old and new code creates a clone.
When `input` is a non-datamodel (a string or path), the old code loads the file as a `RampModel` (no clone is made). The new code loads the file (as whatever model is defined in the file) and then creates a clone (when `model_class(model)` is called).